### PR TITLE
Major (Potential) Update

### PR DESCRIPTION
**Pull request by disciple256**

### CODE REVIEW

**Comment by disciple256** [comment]
Added `/coords`, `/settank`, and `/reset`.
Re-added `/clone` and `/spawn`.
Made a dev tank and structured dev branches.
Made 2 OP tanks and put them in Overdone.
Removed that annoying period.
Did a bunch of other little things.

File: server/lib/definitions.js [hunk 1]
```diff
@@ -4655,4 +4542,319 @@ exports.bot = {
...
+      SPEED: 30,
+      PUSHABILITY: 0,
+    },
+};
+
+var streamProp = { SHOOT_SETTINGS: combineStats([g.basic, g.mini, g.stream, g.halfrecoil]), 
+                   TYPE: exports.bullet, }
+var streamDelay = 0;
+var streamLength = 32; 
+exports.stream2 = {
```
Comment (disciple256) [inline server/lib/definitions.js:4644]: The first OP tank, *OP Streamliner*.
Both OP tanks were created in more or less the same way.

File: server/lib/definitions.js [hunk 2]
```diff
@@ -4655,4 +4542,319 @@ exports.bot = {
...
+            PROPERTIES: streamProp, },
+    ], //25 barrels lol
+};
+
+    streamProp = { SHOOT_SETTINGS: combineStats([g.trap, g.mini, g.stream, g.halfrecoil]), 
+               TYPE: exports.trap, 
+               STAT_CALCULATOR: gunCalcNames.trap, }
+    streamDelay = 0;
+    streamLength = 32; 
+exports.trap2 = {
```
Comment (disciple256) [inline server/lib/definitions.js:4714]: The second OP tank, *OP Barricade*.

File: server/lib/definitions.js [hunk 3]
```diff
@@ -4655,4 +4542,319 @@ exports.bot = {
...
+        POSITION: [streamLength--*2+1,  8,     1,     0,   0,    0,   streamDelay++/25, ], 
+            PROPERTIES: streamProp, }, { 
+        POSITION: [streamLength--*2+1,  8,     1,     0,   0,    0,   streamDelay++/25, ], 
+            PROPERTIES: streamProp, },
+    ], //25 barrels again
+};
+
+//========== UPGRADE PATHS ==========
+
+exports.developer.UPGRADES_TIER_1 = [exports.basic, exports.betaTester, exports.overdone, exports.misc, exports.bosses, exports.observer];
```
Comment (disciple256) [inline server/lib/definitions.js:4781]: Developer upgrades.

File: server/lib/definitions.js [hunk 4]
```diff
@@ -4655,4 +4542,319 @@ exports.bot = {
     AI: { STRAFE: true, },
 };
 
-exports.testbed.UPGRADES_TIER_1.push(exports.elite_sprayer);
\ No newline at end of file
+// DEV TANKS
```
Comment (disciple256) [inline server/lib/definitions.js:4545]: Organized developer branches with 'folders'.

File: clientSrc/app.js [hunk 5]
```diff
@@ -4287,7 +4294,7 @@ const gameDrawDead = (() => {
             'lol you died', x, y - 80, 8, color.guiwhite, 'center'
         );
         text.level.draw(
-            'Level ' + gui.__s.getLevel() + ' ' + mockups[gui.type].name + '.',
+            'Level ' + gui.__s.getLevel() + ' ' + mockups[gui.type].name,
```
Comment (disciple256) [inline clientSrc/app.js:4297]: **BEGONE VILE PERIOD**

File: server/lib/definitions.js [hunk 6]
```diff
@@ -377,7 +377,7 @@ exports.egg = {
```
Comment (disciple256) [inline server/lib/definitions.js:377]: Added *Green* in front of Green Polygons for the sake of the upgrade menu.

File: clientSrc/app.js [hunk 7]
```diff
@@ -742,7 +742,7 @@ global.clickables = (() => {
     })();
     return {
         stat: Region(10),
-        upgrade: Region(8),
+        upgrade: Region(20),
```
Comment (disciple256) [inline clientSrc/app.js:745]: Made more upgrades show up.

File: server/server.js [hunk 8]
```diff
@@ -3527,6 +3527,68 @@ const sockets = (() => {
                               util.warn(`Ran ${args[1]} due to /eval being used in game.`)
                               eval(args[1])
                             break;
+                            case 'summon':
+                            case 'spawn':
```
Comment (disciple256) [inline server/server.js:3531]: Brought back `/spawn` and `/clone`, and added `/coords`, `/settank` and `/reset`.

File: server/lib/definitions.js [hunk 9]
```diff
@@ -4655,4 +4542,319 @@ exports.bot = {
...
+      exports.hexa.UPGRADES_TIER_3 = [exports.octo, exports.hexatrap];
+      exports.auto3.UPGRADES_TIER_3 = [exports.auto5, exports.heavy3, exports.auto4];
+      exports.flanktrap.UPGRADES_TIER_3 = [exports.bushwhack, exports.guntrap, exports.fortress, exports.bomber];
+
+  exports.director.UPGRADES_TIER_2 = [exports.overseer, exports.cruiser, exports.underseer];
+      exports.director.UPGRADES_TIER_3 = [exports.factory];
+      exports.overseer.UPGRADES_TIER_3 = [exports.overlord, exports.overtrap, exports.overgunner];  
+      exports.underseer.UPGRADES_TIER_3 = [exports.necromancer];
+      exports.cruiser.UPGRADES_TIER_3 = [exports.carrier, exports.battleship, exports.fortress];
+
```
Comment (disciple256) [inline server/lib/definitions.js:4836]: Removed `/*` right here and `*/` down there to enable more upgrades.

**Comment by pmagixc** [comment]
no

**Comment by DrakoHyena** [comment]
This has some undesirable features and is not up to date, sorry.

**Comment by disciple256** [comment]
I worked for 6 hours on this :(

**Pull request closed without merging**